cmd/boost/deal_cmd.go

```diff
@@ -0,0 +1,299 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+	"github.com/mitchellh/go-homedir"
+
+	types2 "github.com/filecoin-project/boost/transport/types"
+	"github.com/filecoin-project/go-address"
+	cborutil "github.com/filecoin-project/go-cbor-util"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/api"
+	chain_types "github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+	cli "github.com/urfave/cli/v2"
+
+	inet "github.com/libp2p/go-libp2p-core/network"
+)
+
+const DealProtocolv120 = "/fil/storage/mk/1.2.0"
+
+var dealCmd = &cli.Command{
+	Name:  "deal",
+	Usage: "Make an online deal with Boost",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:     "url",
+			Usage:    "url to CAR file",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "provider",
+			Usage:    "provider address",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "commp",
+			Usage:    "commp of the CAR file",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "piece-size",
+			Usage:    "size of the CAR file as a padded piece",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "car-size",
+			Usage:    "size of the CAR file",
+			Required: true,
+		},
+		&cli.StringFlag{
```

Comment:
Technically `payload-cid` is not required, as we just pass it in the `label` field... so shall we mark it as `optional` ?

Reply:
I think let's mark it as required - even though it's not required technically it's become a kind of defacto standard